tests/testthat/test_change_from_tidy_2_data_structures.R

```diff
@@ -6,29 +6,91 @@ describe("We define the expected data structures", {
   path_coordinates <- "../data/camera_traps_coordinates.csv"
   paths <- list("cameras" = path_cameras, "field" = path_field, "coordinates" = path_coordinates)
   tidy_table <- get_tidy_from_field_and_cameras(paths)
+  capturas <- tidy_table %>%
+    filter(session == 4) %>%
+    group_by(Method) %>%
+    summarize(capturas = sum(r))
   it("Expected Camera-Traps data structure", {
     method <- "Camera-Traps"
-    obtained_final_structure <- tidy_2_final(tidy_table %>% filter(Method == method))
+    final_structure <- Filter_tidy$new(tidy_table)
+    final_structure$select_session(4)
+    final_structure$select_method(method)
+    obtained_final_structure <- final_structure$spatial()
     expected_final_structure <- tibble(
-      Grid = c(38, 15, 31),
+      Grid = c(15, 31, 38),
       session = c(4, 4, 4),
       Method = c("Camera-Traps", "Camera-Traps", "Camera-Traps"),
-      r_1 = c(1, NA, NA),
-      r_2 = c(NA, 1, 0),
-      e_1 = c(1, NA, NA),
-      e_2 = c(NA, 1, 1)
+      r_1 = c(NA, NA, 1),
+      r_2 = c(1, 0, NA),
+      r_3 = as.numeric(NA),
+      r_4 = as.numeric(NA),
+      r_5 = as.numeric(NA),
+      e_1 = c(NA, NA, 1),
+      e_2 = c(1, 1, NA),
+      e_3 = as.numeric(NA),
+      e_4 = as.numeric(NA),
+      e_5 = as.numeric(NA),
     )
     expect_equal(obtained_final_structure, expected_final_structure)
+    expected_capture <- capturas %>%
+      filter(Method == "Camera-Traps") %>%
+      .$capturas
+    obtained_capture <- sum(sum(obtained_final_structure[, 4:8], na.rm = TRUE))
+    expect_equal(obtained_capture, expected_capture)
   })
   it("Expected Hunting data structure", {
     method <- "Hunting"
-    obtained_final_structure <- tidy_2_final(tidy_table %>%
-      arrange(ocassion) %>%
-      filter(Method == method, session == 4)) %>%
-      arrange(Grid)
-
+    final_structure <- Filter_tidy$new(tidy_table)
+    final_structure$select_session(4)
+    final_structure$select_method(method)
+    obtained_final_structure <- final_structure$spatial()
     path_expected <- "../data/hunting_final_structure.csv"
-    expected_final_structure <- read_csv(path_expected)
+    expected_final_structure <- read_csv(path_expected, show_col_types = FALSE, col_types = "nncnnnnnnnnnn")
     expect_equal(obtained_final_structure, expected_final_structure)
+
+    expected_capture <- capturas %>%
+      filter(Method == "Hunting") %>%
+      .$capturas
+    obtained_capture <- sum(sum(obtained_final_structure[, 4:8], na.rm = TRUE))
+    expect_equal(obtained_capture, expected_capture)
+  })
+})
+describe("kdfkjghjkhkjhdfjk", {
```

Comment:
@mvillasante @MemoOlv @nepito:
Could we use a better description than _kdfkjghjkhkjhdfjk_?